Upgrade unity version

aboedo opened the pull request:
Upgrades unity version to `2021.3.3f1` in SubTester and IntegrationTests

vegaro commented:
Yes I agree @aboedo . I tell you is going to be a pita for sure to get the integration tests to work. It took me a while to get them to work in 2019 so I don't expect less with 2021. I bet something has changed

aboedo commented:
I wasted an unfortunate amount of time on this today. I got it close I think, after ssh'ing into the instance I realized that the path to the Unity editor is almost exactly the same as with the other image, it's just `/opt/Unity` -> `/opt/unity`

However I couldn't get the license to work (you need a new one per Unity major version), and I had some trouble requesting a new one in the docker instance ¯\\_(ツ)_/¯

aboedo commented on .circleci/config.yml:
```diff
@@ -58,17 +58,17 @@ commands:
       - run:
           name: Decode license in env variable
-          command: echo $BASE64_LICENSE | base64 --decode >> Unity_v2019.x.ulf
+          command: echo $UNITY_LICENSE | base64 --decode >> Unity.x.ulf
```
I used a separate env var because I didn't want to modify the existing one, that way existing tests don't break

aboedo commented on .circleci/config.yml:
```diff
@@ -39,7 +39,7 @@ commands:
           name: Import package
           command: |
-            /opt/Unity/Editor/Unity \
+            /opt/unity/Editor/Unity \
```
slightly different location in the new image

aboedo commented on IntegrationTests/Assets/Editor/CIEditorScript.cs:
```diff
@@ -119,51 +127,16 @@ public static BuildOptions GetBuildOptions()
-        EditorPrefs.SetString("AndroidSdkRoot", Environment.GetEnvironmentVariable("ANDROID_HOME"));
-        EditorPrefs.SetString("JdkPath", Environment.GetEnvironmentVariable("JAVA_HOME"));
+        EditorPrefs.SetString("AndroidSdkRoot", Environment.GetEnvironmentVariable("ANDROID_HOME_GAME_CI"));
+        EditorPrefs.SetString("JdkPath", Environment.GetEnvironmentVariable("JAVA_HOME_GAME_CI"));
```
I didn't want to mess with existing env vars again

aboedo commented on IntegrationTests/Assets/Editor/CIEditorScript.cs:
```diff
@@ -211,12 +227,87 @@ public static void PerformBuild()
         SetScriptingBackendFromEnv(buildTarget);
 
-        var buildReport = BuildPipeline.BuildPlayer(GetEnabledScenes(), fixedBuildPath, buildTarget, buildOptions);
+        // setting the scene manually because I was getting an error "can't build untitled scene." for iOS
+        string[] scenes = { "Assets/Scenes/Main.unity" };
+        var buildReport = BuildPipeline.BuildPlayer(scenes, fixedBuildPath, buildTarget, buildOptions);
```
this was fun. got build failures for iOS for a long time.

aboedo commented:
changes summary: 

- updated the editor version
- updated to use gameCI https://game.ci/docs/docker/docker-images
- downloaded gameCI image to perform command line arguments to get the license
- with the license, encoded into base64 and stored into env variable
- set env variable for java home and other androidy things https://github.com/game-ci/docker/issues/84
- updated google play services resolver to latest
- updated CIEditorScript to latest version, but kept some workarounds for google resolver
- manually set the Main scene for unity
- updated the bundle id
- updated the unity binary path since it's now lowercase `unity`
- updated the xcode version
- added the license to 1password

aboedo commented:
FINALLY integration tests passing. Marking as ready for review

vegaro reviewed (commented):
I am curious about all the .aar and .jar files. Do we need them in the repo? I think they will get downloaded automatically when installing the EDM4U plugin right?

vegaro commented on IntegrationTests/Assets/ExternalDependencyManager.meta:
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Is the `ExternalDependencyManager` folder is not in the repo so I don't think we need this file

aboedo commented:
I'll try removing them, I think I was just careless and added them all unnecessarily

aboedo commented:
removed the extra files
